src/Microsoft.Extensions.Configuration.FileProviderExtensions/ConfigurationRootExtensions.cs

```diff
@@ -20,8 +21,13 @@ public static IConfigurationRoot ReloadOnChanged(this IConfigurationRoot config,
                 throw new ArgumentNullException(nameof(filename));
             }
 #if NET451
-            var basePath = AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY") as string ??
-                AppDomain.CurrentDomain.BaseDirectory ?? string.Empty;
+            var basePath =
```

Comment:
OT: Yikes, I didn't realize that we had this code duplication. I think this should actually call `FileConfigurationExtensions.GetBasePath()` to resolve the right path if the user has called `configuraitonBuilder.SetBasePath()`. Otherwise this is just looking at the default paths.